test/unit/lib/entities/context/DutchV2QuoteContext.test.ts

```diff
@@ -0,0 +1,687 @@
+import { ID_TO_CHAIN_ID, WRAPPED_NATIVE_CURRENCY } from '@uniswap/smart-order-router';
+import Logger from 'bunyan';
+import { BigNumber, ethers } from 'ethers';
+
+import { NATIVE_ADDRESS, RoutingType } from '../../../../../lib/constants';
+import { DutchQuote, DutchV2QuoteContext } from '../../../../../lib/entities';
+import { SyntheticStatusProvider } from '../../../../../lib/providers';
+import { Erc20__factory } from '../../../../../lib/types/ext/factories/Erc20__factory';
+import {
+  AMOUNT,
+  AMOUNT_GAS_ADJUSTED,
+  AMOUNT_UNDER_GAS_THRESHOLD,
+  CHAIN_OUT_ID,
+  ETH_IN,
+  TOKEN_IN,
+} from '../../../../constants';
+import {
+  BASE_REQUEST_INFO_EXACT_IN,
+  createClassicQuote,
+  createDutchQuote,
+  createDutchQuoteWithRequest,
+  DL_QUOTE_EXACT_IN_BETTER,
+  makeDutchV2Request,
+  QUOTE_REQUEST_DUTCH_V2,
+} from '../../../../utils/fixtures';
+
+describe('DutchV2QuoteContext', () => {
```

Comment:
I would love to reduce some code duplication here but lets take as a follow up

Reply:
yeahh i was thinking the same, and tried a quick deduplicated version but all the type specific fixtures made it v annoying

Reply:
yeah it is v annoying, can look into later